Kernel/FileSystem/FUSE/FUSEConnection.h

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (c) 2024, the SerenityOS developers.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <Kernel/FileSystem/FUSE/Definitions.h>
+#include <Kernel/FileSystem/OpenFileDescription.h>
+#include <Kernel/Library/KBuffer.h>
+
+namespace Kernel {
+
+class FUSEConnection : public RefCounted<FUSEConnection> {
+public:
+    static ErrorOr<NonnullRefPtr<FUSEConnection>> try_create(NonnullRefPtr<OpenFileDescription> description)
+    {
+        if (!description->is_device())
+            return Error::from_errno(EINVAL);
+
+        if (description->device()->class_name() != "FUSEDevice"sv)
+            return Error::from_errno(EINVAL);
+
+        auto* device = bit_cast<FUSEDevice*>(description->device());
+        TRY(device->initialize_instance(*description));
+
+        return TRY(adopt_nonnull_ref_or_enomem(new (nothrow) FUSEConnection(description)));
+    }
+
+    static ErrorOr<NonnullOwnPtr<KBuffer>> create_request(int opcode, int nodeid, u32 unique, ReadonlyBytes request_body)
+    {
+        size_t request_length = sizeof(fuse_in_header) + request_body.size();
+        auto request = TRY(KBuffer::try_create_with_size("FUSE: Request"sv, request_length));
+        memset(request->data(), 0, request_length);
+        fuse_in_header* header = bit_cast<fuse_in_header*>(request->data());
+        header->len = request_length;
+        header->opcode = opcode;
+        header->unique = unique;
+        header->nodeid = nodeid;
+
+        // FIXME: Fill in proper values for these.
+        header->uid = 0;
+        header->gid = 0;
+        header->pid = 1;
+
+        u8* payload = bit_cast<u8*>(request->data() + sizeof(fuse_in_header));
+        memcpy(payload, request_body.data(), request_body.size());
+
+        return request;
+    }
+
+    ErrorOr<NonnullOwnPtr<KBuffer>> send_request_and_wait_for_a_reply(int opcode, int nodeid, ReadonlyBytes request_body)
```

Comment:
I get it that we need to support opcodes that the FUSE framework standardized, but within kernel code it is much more preferable to not pass raw integers. Can we use an `enum` (preferably `class enum`)?

Also, is `nodeid` must be an `int`? what happens if it's negative?

Reply:
The question is not how userspace stores them, but it is safe to pass them. If we check that they're not negative, it is much more easier to pass them as `unsigned` all the way to the actual point we use them, so cognitively we don't have to remember if we deal with negative IDs down the road.